server/src/tests/integration/withdrawal.rs

```diff
@@ -0,0 +1,212 @@
+        "src/tests/res/L2OutputOracle.json"
+    );
+}
+
+mod op_portal {
+    alloy::sol!(
+        #[sol(rpc)]
+        OptimismPortal,
+        "src/tests/res/OptimismPortal.json"
+    );
+}
+
+const WITHDRAW_ADDRESS: Address =
+    alloy::primitives::address!("4200000000000000000000000000000000000016");
+
+pub async fn withdraw_to_l1() -> Result<()> {
+    let amount = "1";
+    let prefunded_wallet = get_prefunded_wallet().await?;
+    let prefunded_address = prefunded_wallet.address();
+
+    // `check_balance=false` balance because tokens are burned.
+    let withdraw_tx_hash =
+        l2_send_ethers(&prefunded_wallet, WITHDRAW_ADDRESS, amount, false).await?;
+
+    let l2_provider = ProviderBuilder::new().on_http(Url::parse(L2_RPC_URL)?);
+    let rx = l2_provider
+        .get_transaction_receipt(withdraw_tx_hash)
+        .await?
+        .unwrap();
+
+    // Extract the withdrawal event from the transaction log
+    let withdrawal_log = rx
+        .inner
+        .logs()
+        .iter()
+        .find(|l| l.address() == WITHDRAW_ADDRESS)
+        .unwrap();
+    let event = withdraw_event();
+    let decoded = event.decode_log(withdrawal_log.data(), true).unwrap();
+    let (withdrawal_hash, _) = decoded.body.last().unwrap().as_fixed_bytes().unwrap();
+
+    // `storage_slot` is calculated based on the Solidity convention for how maps work.
+    let slot_preimage = [withdrawal_hash, &[0u8; 32]].concat();
+    let storage_slot = alloy::primitives::keccak256(slot_preimage);
+
+    let l1_provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(prefunded_wallet))
+        .on_http(Url::parse(&var("L1_RPC_URL")?)?);
+
+    // Contract used on the L1 for withdrawals
+    let portal_address = Address::from_str(&get_deployed_address("OptimismPortalProxy")?)?;
+    let portal_contract = op_portal::OptimismPortal::new(portal_address, &l1_provider);
+
+    // Contract used on the L1 to keep track of the L2 state
+    let oracle_address = Address::from_str(&get_deployed_address("L2OutputOracleProxy")?)?;
+    let l2_oracle_contract = op_oracle::L2OutputOracle::new(oracle_address, &l1_provider);
+
+    // Wait for proposer to push new blocks top L1
+    let withdraw_block_number = withdrawal_log.block_number.unwrap();
+    let now = Instant::now();
+    loop {
+        let l2_block_number: u64 = l2_oracle_contract
+            .latestBlockNumber()
+            .call()
+            .await
+            .unwrap()
+            ._0
+            .saturating_to();
+
+        // Timeout to prevent this from being an infinite loop if something breaks
+        if now.elapsed().as_secs() > 10 * 60 {
+            anyhow::bail!(
+                "WITHDRAW ERROR: L1 contract `L2OutputOracleProxy` not updated to block containing withdraw within 10 minutes. CurrentBlock={l2_block_number} TargetBlock={withdraw_block_number}"
+            );
+        }
+
+        if l2_block_number < withdraw_block_number {
```

Comment:
nit:
```rust
if l2_block_number >= withdraw_block_number {
    break;
}
tokio::time::sleep(Duration::from_secs(OP_BRIDGE_IN_SECS)).await;
```

Reply:
Done in https://github.com/MovedNetwork/op-move/pull/199/commits/58e44ffd7d56df23c0c9f5a0fec846dadf6d2bf2